tools/autograd/templates/Functions.cpp

```diff
@@ -575,6 +575,28 @@ Tensor var_backward(Tensor grad, const Tensor & self, int64_t dim, bool unbiased
   return (2.0 / (self.size(dim) - unbiased)) * grad * (self - self.mean(dim, true));
 }
 
+Tensor std_backward(const Tensor & grad, const Tensor & self, Tensor result, bool unbiased) {
+  Tensor result_zero_mask = (result == 0.);
+  if (result_zero_mask.any().toCByte()) {
```

Comment:
Hmm... in the CUDA case, I wonder if it wouldn't be better to unconditionally apply the masked fill, because that lets you prevent a synchronization.

Reply:
(You didn't specify if your perf experiment was CPU or CUDA)

Reply:
The perf experiment was CPU. Sorry about missing this out.